### DISCUSSION TITLE
Add module delete button

### DISCUSSION THREAD

**Discussion by CqldFire** [NamelessMC/Nameless#2174] [Ideas]
I noticed that the panel templates page and templates page both have the option to delete templates (a button in the template's row), but the module page doesn't have the option to delete modules. I think this should be added as it would be a great addition and would make the delete feature more consistent.

**Comment by samerton** [comment]
The delete button needs to run the onUninstall() method for the module so it can remove any database tables/settings etc

**Comment by tadhgboyle** [comment]
Should this also (attempt) to delete the module's files, as well as calling the `onUninstall()` function?
I think yes

**Comment by CqldFire** [comment]
Yes
